lib/internal/fs.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const Buffer = require('buffer').Buffer;
+const Stream = require('stream').Stream;
+const fs = require('fs');
+
+function assertEncoding(encoding) {
+  if (encoding && !Buffer.isEncoding(encoding)) {
+    throw new Error('Unknown encoding: ' + encoding);
+  }
+}
+
+// Temporary hack for process.stdout and process.stderr when piped to files.
+class SyncWriteStream extends Stream {
+  constructor(fd, options) {
```

Comment:
Default params are not yet fully optimized. However, this class is used at
most once.
On May 14, 2016 1:26 PM, "Jeremiah Senkpiel" <notifications@github.com>
wrote:

> In lib/internal/fs.js
> <https://github.com/nodejs/node/pull/6749#discussion_r63281513>:
>
> > @@ -0,0 +1,73 @@
> > +'use strict';
> > +
> > +const Buffer = require('buffer').Buffer;
> > +const Stream = require('stream').Stream;
> > +const fs = require('fs');
> > +
> > +function assertEncoding(encoding) {
> > +  if (encoding && !Buffer.isEncoding(encoding)) {
> > +    throw new Error('Unknown encoding: ' + encoding);
> > +  }
> > +}
> > +
> > +// Temporary hack for process.stdout and process.stderr when piped to files.
> > +class SyncWriteStream extends Stream {
> > +  constructor(fd, options) {
>
> options = {}? Or have we not checked if default params are speedy enough?
>
> (Quite frankly it doesn't matter here though.)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/6749/files/1173d733a455187d47f4e4eb61a815138e2abfe0#r63281513>
>